packages/components/addon/modifiers/hds-code-mirror.js

```diff
@@ -0,0 +1,124 @@
+// import Modifier, { ArgsFor, PositionalArgs, NamedArgs } from 'ember-modifier';
+import Modifier from 'ember-modifier';
+import { assert } from '@ember/debug';
+import { registerDestructor } from '@ember/destroyable';
+import codemirror from 'codemirror';
+
+// import type Owner from '@ember/owner';
+
+// import 'core/utils/register-codemirror-hcl';
+import 'codemirror/mode/javascript/javascript';
+import 'codemirror/mode/go/go';
+import 'codemirror/mode/markdown/markdown';
+import 'codemirror/mode/shell/shell';
```

Comment:
I'm been trying to think about how to enable consumers to choose what language modes they import but I really don't know how that could work. Do either of you have any recommendations for how that could possibly work or resources I could use to research? @didoo @alex-ju

Reply:
probably I would start by looking at how it's done in the consumers' codebases: how do they do it?

Reply:
Haven't found examples of consumers doing this. They seem to just specify the individual languages they need as part of their components.

Reply:
@KristinLBradley maybe there are interesting code snippets here?
https://github.com/search?q=%22codemirror%2Fmode%22+user%3Ahashicorp&type=code&ref=advsearch

Reply:
Thanks!